src/modules/todoList.js

```diff
@@ -80,6 +80,50 @@ export default class Todo {
     });
   };
 
+  updateStatus = (listIndex, checked) => {
+    const description = document.querySelector(`.desc-${listIndex}`);
+    const listCollection = this.#getLists();
+    let listUpdated = [];
+    if (checked) {
+      description.classList.add('completed');
+      listUpdated = listCollection.map((list) => {
+        if (list.index === listIndex) {
+          return {
+            ...list,
+            completed: true,
+          };
+        }
+
+        return list;
+      });
+    } else {
+      description.classList.remove('completed');
+      listUpdated = listCollection.map((list) => {
+        if (list.index === listIndex) {
+          return {
+            ...list,
+            completed: false,
+          };
+        }
+
+        return list;
+      });
+    }
+    this.#setList(listUpdated);
+  };
+
+  clearCompleted = () => {
+    const listCollection = this.#getLists();
+    const listUpdated = listCollection
+      .filter((list) => !list.completed)
+      .map((list, index) => ({
+        ...list,
+        index: index + 1,
+      }));
+    this.#setList(listUpdated);
+    this.showList();
+  };
```

Comment:
- [ ] The new functionality should be implemented in a new JavaScript file and imported as a module kindly make sure to create a new file for the status updates